python/tank/descriptor/descriptor_config.py

```diff
@@ -71,6 +71,9 @@ def resolve_core_descriptor(self):
         :returns: The core descriptor if :attr:`ConfigDescriptor.associated_core_descriptor` is set,
             ``None`` otherwise.
         """
+        # have to locally import this method due to cyclic deps
+        from .descriptor_factory import create_descriptor
```

Comment:
This didn't use to be an issue when everything was globally imported. Why it is now? Plus, the bug you fixed was introduced by circular deps, might not be good to introduce a new one.

Reply:
there is a cycle in here, so somewhere the has to be a local import, at least given the current way of doing things (without factory registrations etc as you suggest above). I can try that, but i will need to balance this against general complexity. doing a local import of `create_descriptor` is not quite the same thing that we had previously - this is a method and not a class so i don't believe this is an issue in the same way that the classes were.